src/zip.js

```diff
@@ -8,15 +8,23 @@ const { createManifest } = require('./manifest')
 const { getPluginsModulesPath } = require('./node_dependencies')
 const { getFunctionsFromPaths } = require('./runtimes')
 const { ARCHIVE_FORMAT_NONE, ARCHIVE_FORMAT_ZIP } = require('./utils/consts')
-const { listFunctionsDirectories, resolveFunctionsDirectories } = require('./utils/fs')
+const { listFunctionsDirectories, lstat, resolveFunctionsDirectories } = require('./utils/fs')
 const { removeFalsy } = require('./utils/remove_falsy')
 
 const DEFAULT_PARALLEL_LIMIT = 5
 
-const validateArchiveFormat = (archiveFormat) => {
-  if (![ARCHIVE_FORMAT_NONE, ARCHIVE_FORMAT_ZIP].includes(archiveFormat)) {
-    throw new Error(`Invalid archive format: ${archiveFormat}`)
+// Returns the input object with an additional `size` property containing the
+// size of the file at `path` when it is a ZIP archive.
+const addArchiveSize = async (result) => {
+  const { path } = result
+
+  if (extname(path) !== '.zip') {
+    return result
   }
+
+  const { size } = await lstat(path)
```

Comment:
Don't we want to use `fs.stat()` instead of `fs.lstat()` here?
I'm not sure it can actually ever be a symlink there, but if it was, would we want the size of the symlinked path, or the symlinked content?

Reply:
It won't ever be a symlink in this case because we're opening the ZIP file we're creating ourselves, but I agree that there's no reason to use `lstat`. I replaced it with `stat` in https://github.com/netlify/zip-it-and-ship-it/pull/672/commits/6a75c748242bde193d473b2041c4c26fe07dca60.